src/Send/Context/SendContext.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react'
+import {createContext, useContext} from 'react'
+
+import {getDefaultAssetByNetworkId} from '../../legacy/config'
+import {isEmptyString} from '../../legacy/utils'
+import {YoroiWallet} from '../../yoroi-wallets'
+import {TokenId, YoroiAmounts} from '../../yoroi-wallets/types'
+
+type SendProvider = {
+  wallet: YoroiWallet
+  balances: YoroiAmounts
+}
+
+const SendContext = createContext<undefined | SendContext>(undefined)
+export const SendProvider: React.FC<SendProvider> = ({children, balances, wallet}) => {
+  const primaryTokenId = getDefaultAssetByNetworkId(wallet.networkId).identifier
+
+  const [state, dispatch] = React.useReducer(sendReducer, {...initialState, selectedTokenId: primaryTokenId})
+  const sendActions = sendActionsMapper(dispatch)
+
+  React.useEffect(() => {
```

Comment:
im on the fence about this.
its a hack, but it doesnt seem horrible
see if you can make it work inline

Reply:
Done